enterprise/frontend/src/metabase-enterprise/audit_app/components/InstanceAnalyticsButton/InstanceAnalyticsButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { t } from "ttag";
+import { useEffect } from "react";
+import { push } from "react-router-redux";
+import { loadInfo } from "metabase-enterprise/audit_app/reducer";
+import EntityMenuItem from "metabase/components/EntityMenuItem";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import type { AuditInfoState } from "metabase-enterprise/audit_app/types/state";
+import type { DashboardId } from "metabase-types/api";
+
+interface InstanceAnalyticsButtonProps {
+  entitySelector: (state: AuditInfoState) => number;
+  linkQueryParams: { dashboard_id: DashboardId } | { question_id: number };
+}
+
+export const InstanceAnalyticsButton = ({
+  entitySelector,
+  linkQueryParams,
+}: InstanceAnalyticsButtonProps) => {
+  const dispatch = useDispatch();
+  const entityId = useSelector(state =>
+    entitySelector(state as AuditInfoState),
+  );
+
+  useEffect(() => {
+    dispatch(loadInfo());
+  }, [dispatch]);
+
+  if (entityId !== undefined) {
```

Comment:
I would maybe loosen up this check to `!entityId`

Reply:
My fear here was IDs that are 0, though technically that shouldn't ever happen